instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -508,14 +695,92 @@ def _uninstrument(self, **kwargs: Any):
             unwrap(redis.client.Pipeline, "immediate_execute_command")
-        if redis.VERSION >= _REDIS_CLUSTER_VERSION:
+        if _CLIENT_CLUSTER_SUPPORT:
             unwrap(redis.cluster.RedisCluster, "execute_command")
             unwrap(redis.cluster.ClusterPipeline, "execute")
-        if redis.VERSION >= _REDIS_ASYNCIO_VERSION:
+        if _CLIENT_ASYNCIO_SUPPORT:
             unwrap(redis.asyncio.Redis, "execute_command")
             unwrap(redis.asyncio.Redis, "pipeline")
             unwrap(redis.asyncio.client.Pipeline, "execute")
             unwrap(redis.asyncio.client.Pipeline, "immediate_execute_command")
-        if redis.VERSION >= _REDIS_ASYNCIO_CLUSTER_VERSION:
+        if _CLIENT_ASYNCIO_CLUSTER_SUPPORT:
             unwrap(redis.asyncio.cluster.RedisCluster, "execute_command")
             unwrap(redis.asyncio.cluster.ClusterPipeline, "execute")
+
+    @staticmethod
+    def instrument_client(
+        client: redis.StrictRedis
+        | redis.Redis
+        | redis.asyncio.Redis
+        | redis.cluster.RedisCluster
+        | redis.asyncio.cluster.RedisCluster,
+        tracer_provider: TracerProvider | None = None,
+        request_hook: RequestHook | None = None,
+        response_hook: ResponseHook | None = None,
+    ):
+        """Instrument the provided Redis Client. The client can be sync or async.
+        Cluster client is also supported.
+
+        Args:
+            client: The redis client.
+            tracer_provider: A TracerProvider, defaults to global.
+            request_hook: a function with extra user-defined logic to run before
+                performing the request.
+
+                The ``args`` is a tuple, where items are
+                command arguments. For example ``client.set("mykey", "value", ex=5)`` would
+                have ``args`` as ``('SET', 'mykey', 'value', 'EX', 5)``.
+
+                The ``kwargs`` represents occasional ``options`` passed by redis. For example,
+                if you use ``client.set("mykey", "value", get=True)``, the ``kwargs`` would be
+                ``{'get': True}``.
+
+            response_hook: a function with extra user-defined logic to run after
+                the request is complete.
+
+                The ``args`` represents the response.
+        """
+        if not hasattr(client, INSTRUMENTATION_ATTR):
+            setattr(client, INSTRUMENTATION_ATTR, False)
+        if not getattr(client, INSTRUMENTATION_ATTR):
+            _instrument_client(
+                client,
+                RedisInstrumentor._get_tracer(tracer_provider=tracer_provider),
+                request_hook=request_hook,
+                response_hook=response_hook,
+            )
+            setattr(client, INSTRUMENTATION_ATTR, True)
+        else:
+            _logger.warning(
+                "Attempting to instrument Redis connection while already instrumented"
+            )
+
+    @staticmethod
+    def uninstrument_client(
```

Comment:
Isn't _instrument_client instrumenting more methods that the two unwrapped here?